Build failure: kermit

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

nix build github:nixos/nixpkgs/0196c0175e9191c474c26ab5548db27ef5d34b05#kermit

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/8ww279rrk2w1vqrdsh88pg0hfypgmqaa-kermit-9.0.302.drv' failed with exit code 2;
       last 25 log lines:
       > grep: /usr/include/features.h: No such file or directory
       > make[1]: Entering directory '/build/src'
       > Making C-Kermit "9.0.302" for Linux 1.2 or later...
       > IMPORTANT: Read the comments in the linux section of the
       > makefile if you have trouble.
       > make xermit KTARGET=${KTARGET:-linuxa} "CC = gcc" "CC2 = gcc" \
       > "CFLAGS = -O -DLINUX -pipe -funsigned-char -DFNFLOAT -DCK_POSIX_SIG \
       > -DCK_NEWTERM -DTCPSOCKET -DLINUXFSSTND -DNOCOTFMC -DPOSIX \
       > -DUSE_STRERROR -D_IO_file_flags" "LNKFLAGS = -lcrypt -lresolv" \
       > "LIBS = -lutil    -lm"
       > make[2]: Entering directory '/build/src'
       > gcc -O -DLINUX -pipe -funsigned-char -DFNFLOAT -DCK_POSIX_SIG -DCK_NEWTERM -DTCPSOCKET -DLINUXFSSTND -DNOCOTFMC -DPOSIX -DUSE_STRERROR -D_IO_file_flags -DKTARGET=\"linux\" -c ckcmai.c
       > ckcmai.c: In function 'main':
       > ckcmai.c:3160:13: error: implicit declaration of function 'time' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]
       >  3160 |         c = time(NULL);
       >       |             ^~~~
       > ckcmai.c:1587:1: note: 'time' is defined in header '<time.h>'; this is probably fixable by adding '#include <time.h>'
       >  1586 | #include "ckcsig.h"
       >   +++ |+#include <time.h>
       >  1587 | ckjmpbuf cmjbuf;
       > make[2]: *** [makefile:985: ckcmai.o] Error 1
       > make[2]: Leaving directory '/build/src'
       > make[1]: *** [makefile:6054: linuxa] Error 2
       > make[1]: Leaving directory '/build/src'
       > make: *** [makefile:6091: linux] Error 2
       For full logs, run 'nix log /nix/store/8ww279rrk2w1vqrdsh88pg0hfypgmqaa-kermit-9.0.302.drv'.
error: 1 dependencies of derivation '/nix/store/g85nzl3naqs08jm0ai2mcmkzn1l1v30z-nix-shell-env.drv' failed to build
```

### Additional context

Following build failure of kermit was against nixos-unstable (0196c0175e9191c474c26ab5548db27ef5d34b05).

We probably should consider moving to the `10.0 Beta.11` which is fairly recent (2024): https://www.kermitproject.org/ckupdates.html since the existing 9.0 version we are on is from 2011.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 24.11 (Vicuna), 24.11.20241130.62c435d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"nixos-22.05.3300.82379884b2e"`
 - nixpkgs: `/nix/store/ljk3wvwqqzbmz0pys49ii18qmsi5aibc-source`

### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ke/kermit/package.nix
+++ b/pkgs/by-name/ke/kermit/package.nix
@@ -39,6 +39,8 @@ stdenv.mkDerivation {
     make -f makefile install
   '';
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-implicit-function-declaration -Wno-implicit-int";
+
   meta = with lib; {
     homepage = "https://www.kermitproject.org/ck90.html";
     description = "Portable Scriptable Network and Serial Communication Software";

